core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandler.java

```diff
@@ -356,11 +359,34 @@ private void setFinalResult(
               TimeUnit.NANOSECONDS);
         }
       }
+      // log the warnings if they have NOT been disabled
+      if (LOG.isWarnEnabled()
```

Comment:
is there an option for this `LOG.isWarnEnabled()` to be changed dynamically?
If not then I think that we can evaluate those conditions only once and assign to a field (to avoid re-evaluation for every request)

Reply:
It is possible to change the log level of a given logger in client code (I don't think any of the driver code currently does this). I'm not sure how often, if ever, it is done in production though.

Reply:
We should assume that log levels can change at runtime. I don't know how common it is in practice, but it definitely makes sense to have a way to adjust them to debug integration/production issues (I would build that into a client application).